### PR TITLE
test(results): cover explorer staging cleanup failures

### PR DESCRIPTION
**Pull request by joeharris76**
## Summary

- Add fault-injection coverage for every staged Explorer pipeline failure boundary.
- Verify the original exception remains visible and the exact temporary staging directory is removed.

## Verification

- `uv run -- python -m pytest tests/unit/scripts/explorer_pipeline -q` (345 passed)
- `make pr-preflight` (27,773 passed, 19 skipped, 4 subtests)

### CODE REVIEW

**Comment by joeharris76** [comment]
@codex please review

File: tests/unit/scripts/explorer_pipeline/test_pipeline.py [hunk 1]
```diff
@@ -919,6 +919,11 @@ def test_benchmark_rankings_short_id_matches_short_ids_table(self, data_dir: Pat
 
 
 class TestStagedOutputGuards:
+    @staticmethod
+    def _assert_no_staging_leftovers(output: Path) -> None:
```
[the remark this answers is not in the document]
Comment (joeharris76) [reply tests/unit/scripts/explorer_pipeline/test_pipeline.py:924]: Rechecked the authoritative Git commit object after rebasing onto current develop. Head 4b239612ea1d417fc23b1ba571ff120ed4aa750d records Joe Harris <joeharris76@gmail.com> as both author and committer. The conflict-resolved branch also passes the focused 68-test file, Ruff, duplicate delta, and full preflight (27,800 passed). Resolving this stale identity finding.
